lib/src/barback/asset_environment.dart

```diff
@@ -167,21 +176,42 @@ class AssetEnvironment {
   /// go to barback immediately.
   Set<AssetId> _modifiedSources;
 
-  AssetEnvironment._(this.graph, this.barback, this.mode, this._watcherType,
-      this._hostname, this._basePort, this.environmentConstants);
+  /// The compiler mode for this environment.
+  final CompilerMode compilerMode;
+
+  AssetEnvironment._(
+      this.graph,
+      this.barback,
+      this.mode,
+      this._watcherType,
+      this._hostname,
+      this._basePort,
+      this.environmentConstants,
+      this.compilerMode);
 
-  /// Gets the built-in [Transformer]s that should be added to [package].
+  /// Gets the built-in [Transformer]s or [AggregateTransformer]s that should be
+  /// added to [package].
   ///
   /// Returns `null` if there are none.
-  Iterable<Transformer> getBuiltInTransformers(Package package) {
-    // Built-in transformers only apply to the root package.
-    if (package.name != rootPackage.name) return null;
+  Iterable<Set> getBuiltInTransformers(Package package) {
+    var transformers = <Set>[];
+    // Transitive built-in transformers apply to all packages.
+    //
+    // Today these transformers are only for dartdevc.
```

Comment:
[nit] this seems like the kind of comment that is very unlikely to be kept up to date if the information goes stale... should we omit it?